org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoProductWindow.java

```diff
@@ -636,15 +649,21 @@ protected void refresh(int M_Warehouse_ID, int M_PriceList_Version_ID)
 		}
 	}	//	refresh
 	
-	// Elaine 2008/11/26
-	/**
-	 *	Query ATP
-	 * @param m_M_Warehouse_ID
-	 * @param m_M_Product_ID 
-	 */
-	protected void initAtpTab (int  m_M_Warehouse_ID, int m_M_Product_ID)
+	protected ColumnInfo[] getAtpColumns()
+	{
+       return new ColumnInfo[]{ new ColumnInfo(Msg.translate(Env.getCtx(), "Date"), "Date", String.class, true, "Date"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "QtyOnHand"), "QtyOnHand", Double.class, true, "QtyOnHand"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "C_BPartner_ID"), "C_BPartner_ID", Double.class, true, "C_BPartner_ID"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "QtyOrdered"), "QtyOrdered", Double.class, true, "QtyOrdered"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "QtyReserved"), "QtyReserved", Double.class, true, "QtyReserved"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "M_Locator_ID"), "M_Locator_ID", Double.class, true, "M_Locator_ID"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "M_AttributeSetInstance_ID"), "M_AttributeSetInstance_ID", Double.class, true, "M_AttributeSetInstance_ID"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "DocumentNo"), "DocumentNo", Double.class, true, "DocumentNo"),
+                                new ColumnInfo(Msg.translate(Env.getCtx(), "M_Warehouse_ID"), "M_Warehouse_ID", Double.class, true, "M_Warehouse_ID")};
+    }
+	
+	protected Vector<String> getAtpLayout()
```

Comment:
The true is for read only.

Reply:
ok, I misread that but still since the readonly changes is not related to the objective of this ticket (save column width), it should not be part of this PR.